src/app/public/modules/lookup/lookup.service.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Injectable
+} from '@angular/core';
+
+/**
+ * @internal
+ */
+@Injectable()
+export class SkyLookupService {
+
+  public isEquivalent(object1: any, object2: any): boolean {
```

Comment:
I'm not sure about this implementation; it looks to only check the top-level properties, so if you had a property of a property that was not equal, then you'd get a false positive. It also assumes the object returned in the add callback will have all the same properties as the one in the data source, but they might be different for a good reason, like saving bits when getting the list for the data source. Can we be more deterministic, like ask the consumer for an `id` field that we can use to compare? Or could we just require that the object passed as the `item` parameter is the same instance as the one in the `data` array so that we can just do a simple equality comparison?

Reply:
Ya. I agree. This was some carry over from the early days of this project. It had actually been bothering me too overnight as I thought more about the fact that this function had taken off more than I originally envisioned. 

Unfortunately, we are a little tied down with the implementation here regarding items in the data source not being required to have an ID. I don't want to change that as they aren't required today and that would be potentially a breaking change. We were actually using this check a few places throughout the show more modal stuff. 

I tracked this down today and it appears that the function was necessary because of cloning of the items that we were doing. However, that cloning wasn't really needed as we aren't changing anything about the data and so I was able to remove that cloning and we'll just go forward with requiring the object that is returned to the callback function to be the same one as the one in the data source. All other places where this function was used can change to a simple equality check just with us removing our internal cloning.